File: Algorithms/Easy/running_sum_1d.py
from typing import List

class Solution:
    def runningSum(self, nums: List[int]) -> List[int]:
        
        s = nums[0]
        running_sum = [nums[0]]
        i = 0
        while i < len(nums) - 1:
            s += nums[i + 1]
            running_sum.append(s)
            i += 1
        
        return running_sum

File: Algorithms/Easy/running_sum_1d.cpp
class Solution {
public:
    vector<int> runningSum(vector<int>& nums) {
        
        const size_t dim = nums.size();
        vector<int> running_sum(dim);

        size_t sum = nums.front();
        running_sum[0] = sum;

        for (size_t i = 1; i != nums.size(); ++i) {
            sum += nums[i];
            running_sum[i] = sum;
        }

        return running_sum;

    }
};
